axelrod/player.py

```diff
@@ -87,10 +88,27 @@ class Player(object):
     def __new__(cls, *args, **kwargs):
         """Caches arguments for Player cloning."""
         obj = super().__new__(cls)
-        obj.init_args = args
-        obj.init_kwargs = kwargs
+        obj.init_kwargs = cls.init_params(*args, **kwargs)
         return obj
 
+    @classmethod
+    def init_params(cls, *args, **kwargs):
+        """
+        Return a dictionary containing the init parameters of a strategy (without 'self').
+        Use *args and *kwargs as value if specified
+        and complete the rest with the default values.
+        """
+        sig = inspect.signature(cls.__init__)
+        # the 'self' parameter needs to be removed or the first *args will be assigned to it
+        self_param = sig.parameters.get('self', None)
+        if self_param is not None:
```

Comment:
There **should** always be a 'self' parameters to get rid of, but since I'm calling `list.remove` which would fail if I call it with None, better be safe than sorry I guess. But I agree that there is no reason an `__init__` would ever have no 'self' parameter so I guess I could remove this test.